observations/obs_converters/AIRS/amsua_netCDF_support_mod.f90

```diff
@@ -1090,6 +1097,112 @@ subroutine fill_amsua_variables(granule, ncid, context)
 end subroutine fill_amsua_variables
 
 
+!------------------------------------------------------------------------------
+!> Determine max possible obs from all files. 
+!> Allocating way too much results in excessive run-time and memory use.
+
+function max_possible_obs(num_files, scan_thin, xtrack_thin, channels)
+
+integer, intent(in) :: num_files
+integer, intent(in) :: scan_thin
+integer, intent(in) :: xtrack_thin
+logical, intent(in) :: channels(:)
+integer :: max_possible_obs
+
+integer :: shrinkage, num_channels, ichan
+
+shrinkage    = 1  ! no thinning
+
+num_channels = 0
+do ichan = 1, size(channels)
+   if (channels(ichan)) num_channels = num_channels + 1
+enddo
+
+if (  scan_thin > 0) shrinkage = shrinkage *   scan_thin 
+if (xtrack_thin > 0) shrinkage = shrinkage * xtrack_thin
+
+max_possible_obs = AMSUA_BT_GEOXTRACK * AMSUA_BT_GEOTRACK * num_channels * &
+                   num_files / shrinkage
+
+end function max_possible_obs
+
+
+!-------------------------------------------------------------------------------
+!> Given two observation sequences, insert the observations from the first
+!> sequence into the second sequence.
+
+function combine_sequences(seq_in, seq_out, filename)
```

Comment:
i'm a little concerned about the performance of this approach.  as the output files get larger the cost of inserting an obs in the file grows.  the original code added new obs to the eventual output sequence.  this version creates a separate sequence (one insert here) and then combines them (another insert).  could this code pass the eventual output sequence into the fill routine and have it put the obs where they are eventually going to be?

Reply:
I understand ... mostly.  The separate obs sequence (the one for each file) is essentially guaranteed to be an append for each observation - not a searching insert. That should be about as fast as possible.  By creating a separate sequence for each file, it is possible to report on the full characteristics of THAT sequence, as opposed to the incremental change to the output sequence. Maybe that's not worthwhile ... dunno. 
The combination insert is where the inefficiency (if any) lies. If the user (and that's a big if) puts the input files in temporally ascending order, these are all appends - again - about as efficient as possible. If the files listed are out-of-order, they are their own worst enemy to begin with. 
The logic of the code got a bit confusing when I tried to use a single sequence, and had a pretty large footprint on the code.

Reply:
i might assume that someone who wanted to print detailed output about the input files might use a different program (obs_info, for example).  printing 240 detailed summaries in a single output file seems hard to find what you are looking for.   but i can see the logic in making separate sequences and concatinating them.  so as long as the performance is ok then this seems ok to me.